gcp/appengine/run_tests.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash -ex
+
+python3 -m pipenv sync
```

Comment:
Comparing with docker/worker/run_tests.sh, can you make this rely on the system installed `pipenv` instead of the default python3's pipenv module?

I'm having [problems with gLinux's default python 3.10](https://github.com/pypa/pipenv/issues/5088) and found the most surgical fix for now is to modify the shebang of `/usr/bin/pipenv` to explicitly use python 3.9

Alternatively, allow overriding the Python to use with an environment variable. Either way, the various run_tests.sh are in an inconsistent state right now...

Reply:
I typically invoke this through my "root" pipenv from the `osv` directory, where I have control over the Python version. i.e.

```python
$ cd osv.dev
$ pipenv shell  # previously set up to be Python 3.9
$ cd gcp/appengine
$ ./run_tests.sh  # python3 -> python3.9 because of the root pipenv
```

Does that work for you?

Reply:
In theory that might if I've curated a pipenv environment accordingly. It does not currently:

```
/usr/local/google/home/apollock/.local/share/virtualenvs/osv.dev-cTOZqQPe/bin/python3: No module named pipenv
```

If I'm outside of a pipenv and run say `make worker-tests`, I have a bad time:

```
$ make appengine-tests
cd gcp/appengine && ./run_tests.sh
+ python3 -m pipenv sync
Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/lib/python3/dist-packages/pipenv/__main__.py", line 4, in <module>
    cli()
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 696, in main
    with self.make_context(prog_name, args, **extra) as ctx:
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 621, in make_context
    self.parse_args(ctx, args)
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 1018, in parse_args
    rest = Command.parse_args(self, ctx, args)
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 875, in parse_args
    parser = self.make_parser(ctx)
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 821, in make_parser
    for param in self.get_params(ctx):
  File "/usr/lib/python3/dist-packages/pipenv/vendor/click/core.py", line 774, in get_params
    help_option = self.get_help_option(ctx)
  File "/usr/lib/python3/dist-packages/pipenv/cli.py", line 26, in get_help_option
    from .import core
  File "/usr/lib/python3/dist-packages/pipenv/core.py", line 21, in <module>
    import requests
  File "/usr/lib/python3/dist-packages/pipenv/vendor/requests/__init__.py", line 65, in <module>
    from . import utils
  File "/usr/lib/python3/dist-packages/pipenv/vendor/requests/utils.py", line 27, in <module>
    from .cookies import RequestsCookieJar, cookiejar_from_dict
  File "/usr/lib/python3/dist-packages/pipenv/vendor/requests/cookies.py", line 172, in <module>
    class RequestsCookieJar(cookielib.CookieJar, collections.MutableMapping):
AttributeError: module 'collections' has no attribute 'MutableMapping'
make: *** [Makefile:24: appengine-tests] Error 1
```

Reply:
What if you just install `pipenv` in the root pipenv environment? This is a dev package specified already here: https://github.com/google/osv.dev/blob/989c3a8b1d1398a95c858a1c4af0639c1f23715a/Pipfile#L13

Reply:
pipenv inside pipenv. Wild. Yes, that seems to work.

So is the intention for developers to basically be preexisting inside a `pipenv shell` instance for Makefile-based operations? That was unclear to me.

Reply:
Yeah, the root pipenv also includes dev tools like pylint and yapf. 

https://github.com/google/osv.dev/blob/master/CONTRIBUTING.md#contributing-code has `pipenv shell` as one of the prereqs, but it probably needs to be made clearer. Feel free to suggest a PR!